src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -78,33 +102,539 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
+
+            List<Entry> entries = file.getEntry();
+            HashMap<String, String> fields = new HashMap<>();
+
+            for (Entry entry : entries) {
+                BibEntry bibEntry = new BibEntry(DEFAULT_BIBTEXENTRY_ID);
+                if (entry.getArticle() != null) {
+                    bibEntry.setType("article");
+                    parseArticle(entry.getArticle(), fields);
+                }
+                if (entry.getBook() != null) {
+                    bibEntry.setType("book");
+                    parseBook(entry.getBook(), fields);
+                }
+                if (entry.getBooklet() != null) {
+                    bibEntry.setType("booklet");
+                    parseBooklet(entry.getBooklet(), fields);
+                }
+                if (entry.getConference() != null) {
+                    bibEntry.setType("conference");
+                    parseConference(entry.getConference(), fields);
+                }
+                if (entry.getInbook() != null) {
+                    bibEntry.setType("inbook");
+                    parseInbook(entry.getInbook(), fields);
+                }
+                if (entry.getIncollection() != null) {
+                    bibEntry.setType("incollection");
+                    parseIncollection(entry.getIncollection(), fields);
+                }
+                if (entry.getInproceedings() != null) {
+                    bibEntry.setType("inproceedings");
+                    parseInproceedings(entry.getInproceedings(), fields);
+                }
+                if (entry.getManual() != null) {
+                    bibEntry.setType("manual");
+                    parseManual(entry.getManual(), fields);
+                }
+                if (entry.getMastersthesis() != null) {
+                    bibEntry.setType("mastersthesis");
+                    parseMasterthesis(entry.getMastersthesis(), fields);
+                }
+                if (entry.getMisc() != null) {
+                    bibEntry.setType("misc");
+                    parseMisc(entry.getMisc(), fields);
+                }
+                if (entry.getPhdthesis() != null) {
+                    bibEntry.setType("phdthesis");
+                    parsePhdthesis(entry.getPhdthesis(), fields);
+                }
+                if (entry.getProceedings() != null) {
+                    bibEntry.setType("proceedings");
+                    parseProceedings(entry.getProceedings(), fields);
+                }
+                if (entry.getTechreport() != null) {
+                    bibEntry.setType("techreport");
+                    parseTechReport(entry.getTechreport(), fields);
+                }
+                if (entry.getUnpublished() != null) {
+                    bibEntry.setType("unpublished");
+                    parseUnpublished(entry.getUnpublished(), fields);
+                }
+                if (entry.getId() != null) {
```

Comment:
This if condition shouldn't be necessary as the xsd schema file specifies the attribute as `use="required"`